File: 0129-sum-root-to-leaf-numbers/0129-sum-root-to-leaf-numbers.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def sumNumbers(self, root: Optional[TreeNode]) -> int:
        def solve(root,num):
            if root.left==None and root.right==None:
                return num*10+root.val
            else:
                a=b=0
                if root.left:
                    a=solve(root.left,num*10+root.val)
                if root.right:
                    b=solve(root.right,num*10+root.val)
                return a+b
        if root is None:
            return 0
        return solve(root,0)
        

File: 0129-sum-root-to-leaf-numbers/0129-sum-root-to-leaf-numbers.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:
    int ans=0;
    int sumNumbers(TreeNode* root) {
        if(root==nullptr)return 0;
        solve(root,0);
        return ans;
    }
private:
  void solve(TreeNode* root,int curr)
  {
        if(root->left==nullptr and root->right==nullptr){ans+=curr*10+root->val;return;}
      curr=curr*10+root->val;
      if(root->left)solve(root->left,curr);
      if(root->right)solve(root->right,curr);
  }
};